File: 프로그래머스/lv1/장정윤/신고 결과 받기.py
def solution(id_list, report, k):
    answer = [0] * len(id_list)
    cnt = [0] * len(id_list)
    hash_map = {id_list[i]: i for i in range(len(id_list))}
    attack = []
    receive = []

    report = list(set(report))

    for rep in report:
        reportList = rep.split(' ')
        attack.append(reportList[0])
        receive.append(reportList[1])
        cnt[hash_map[reportList[1]]] += 1

    return answer

File: 프로그래머스/lv1/이선영/신고 결과 받기.java
import java.util.*;
class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        HashMap<String, Integer> map = new HashMap<>();
        List<String>[] list = new ArrayList[id_list.length];
        
        for(int i=0; i<id_list.length; i++){
            list[i] = new ArrayList<>();
            map.put(id_list[i], i);
        }
        
        for(int i=0; i<report.length; i++){
            String[] rpt = report[i].split(" ");
            int sender = map.get(rpt[0]);
            int receiver = map.get(rpt[1]);
            
            if(!list[receiver].contains(rpt[0])){
                list[receiver].add(rpt[0]);
            }    
        }
        
        for(int i=0; i<id_list.length; i++){
            int idx = map.get(id_list[i]);// 사람 인덱스
            if(list[idx].size()>=k){
                for(String user : list[idx]){
                    
                    int userIdx = map.get(user);
                    answer[userIdx]++;
                }
            }
            
        }
        
        return answer;
    }
}
